api/src/main/java/org/commonjava/indy/data/DefaultStoreValidator.java

```diff
@@ -0,0 +1,173 @@
+import org.commonjava.indy.IndyWorkflowException;
+import org.commonjava.indy.conf.IndyConfiguration;
+import org.commonjava.indy.model.core.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+
+public class DefaultStoreValidator implements StoreValidator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultStoreValidator.class);
+
+    @Inject
+    @WeftManaged
+    @ExecutorConfig( named="store-validation", threads=2, priority=6 )
+    private ExecutorService executorService;
+
+    @Inject
+    IndyConfiguration configuration;
+
+    @Override
+    public ArtifactStoreValidateData validate(ArtifactStore artifactStore) throws InvalidArtifactStoreException, MalformedURLException {
+        final CountDownLatch httpRequestsLatch = new CountDownLatch(2);
+        final HashMap<String, String> errors = new HashMap<>();
+        Optional<URL> remoteUrl = Optional.empty();
+
+        try {
+            if(StoreType.remote == artifactStore.getType()) {
+                // Cast to Remote Repository
+                RemoteRepository remoteRepository = (RemoteRepository) artifactStore;
+                boolean disabled = remoteRepository.isDisabled();
+                if(disabled) {
+                    throw new InvalidArtifactStoreException("Disabled Store", null, null);
+                }
+                //Validate URL from remote Repository URL
+                remoteUrl = Optional.of(new URL(remoteRepository.getUrl()));
+                // Check if remote.ssl.required is set to true and that remote repository protocol is https = throw IndyArtifactStoreException
+                if(configuration.isSSLRequired()
+                    && !remoteUrl.get().getProtocol().equalsIgnoreCase(StoreValidationConstants.HTTPS)) {
+
+                    //Check First if this remote repository is in allowed repositories from remote.nossl.hosts Config Variable
+                    List<String> remoteNoSSLHosts = configuration.getRemoteNoSSLHosts();
+                    String host = remoteUrl.get().getHost();
+
+                    for(String remoteHost : remoteNoSSLHosts) {
+                        LOGGER.warn(
+                            " RemoteHost " +host+ " in allowed list of remote hosts " + remoteNoSSLHosts + " is " +
+                                allowedNonSSLHostname(remoteHost, host) + " allowed " +
+                                " Remote URL: " + remoteUrl.get().toString()
+                        );
+                        // .apache.org , 10.192. , .maven.redhat.com
+                        if(allowedNonSSLHostname(remoteHost,host)) {
```

Comment:
I'm still not sure we're capturing negative results here, and returning a negative validation result. I think we need a boolean defined outside the loop, set to false, then as we iterate through remoteNoSSLHosts, if any rule passes then we can set the boolean to true and break the for loop.

We also need a way to keep track of the overall validation status...that would be a separate boolean for the whole validation process. This one would default to true, I think, and any failing validation rule would set it to false. But we might not want to short-circuit the validation process. That way we get as much validation information as we can before we return...which helps to reduce trial-and-error on the user's part. Obviously this won't work as well for exceptions, but we should try to do this as much as we can.

The main point is that we need to track validation status so we can return a clear signal to the user.

Reply:
Yes , you are right , this is not a way to do it. I will re wright it to accept cases as "*.redhat.com" for example , where "*" is declaring that all subdomains from that domain should be accepted.

Reply:
This is done also , right now i have new logic for checking all allowed domain names which might include "*" for allowing different subdomains.
Maybe it will be better if this logic we transfer it in new bean from which we can inspect all cases where is needed .